packages/shared/src/hooks/post/useTrackedCopyPostLink.ts

```diff
@@ -1,23 +1,40 @@
-import { useContext } from 'react';
+import { useCallback, useContext } from 'react';
 import { Post } from '../../graphql/posts';
 import { useCopyPostLink } from '../useCopyPostLink';
 import AnalyticsContext from '../../contexts/AnalyticsContext';
 import { ShareProvider } from '../../lib/share';
 import { postAnalyticsEvent } from '../../lib/feed';
 import { Origin } from '../../lib/analytics';
+import { useTrackedLink } from '../utils/useTrackedLink';
+import { ReferralCampaignKey } from '../../lib/referral';
 
-type UseTrackedCopyPostLink = (provider: ShareProvider) => void;
+interface UseTrackedCopyPostLink {
+  onCopyLink: (provider: ShareProvider) => void;
+  isLoading: boolean;
+}
 
-export const useTrackedCopyPostLink = (post: Post): UseTrackedCopyPostLink => {
-  const [, copyLink] = useCopyPostLink(post.commentsPermalink);
+export const useTrackedCopyPostLink = (
+  post: Post,
+  origin = Origin.FeedbackCard,
+): UseTrackedCopyPostLink => {
+  const { shareLink, isLoading } = useTrackedLink({
+    link: post.commentsPermalink,
+    cid: ReferralCampaignKey.SharePost,
+  });
+  const [, copyLink] = useCopyPostLink(shareLink);
   const { trackEvent } = useContext(AnalyticsContext);
 
-  return (provider: ShareProvider) => {
```

Comment:
The only thing that changed here was the `origin` and had to put it in a `useCallback`.